adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/lwm2m/LwM2MResourceDirectory.java

```diff
@@ -0,0 +1,694 @@
+            final BindingMode bindingMode,
+            final Long lifetime,
+            final Link[] objectlinks,
+            final String registrationId) {
+
+        final Promise<ResponseCode> result = Promise.promise();
+
+        final Span currentSpan = TracingHelper
+                .buildChildSpan(getTracer(), ctx.getTracingContext(), "update LwM2M device registration", getAdapter().getTypeName())
+                .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_SERVER)
+                .withTag(TracingHelper.TAG_TENANT_ID, ctx.getTenantId())
+                .withTag(TracingHelper.TAG_DEVICE_ID, ctx.getOriginDevice().getDeviceId())
+                .withTag(TAG_LWM2M_REGISTRATION_ID, registrationId)
+                .start();
+
+        LOG.debug("processing LwM2M registration update request [tenant-id: {}, device-id: {}, registration ID]",
+                ctx.getTenantId(), ctx.getOriginDevice().getDeviceId(), registrationId);
+        final Map<String, Object> items = new HashMap<>(2);
+        Optional.ofNullable(bindingMode).ifPresent(bm -> items.put("binding mode", bm.name()));
+        Optional.ofNullable(lifetime).ifPresent(lt -> items.put("lifetime (secs)", lt));
+        currentSpan.log(items);
+
+        if (lifetime != null && lifetime <= 0) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST,
+                    "lifetime query parameter must be positive integer"));
+        } else if (bindingMode != null && !supportedBindingModes.contains(bindingMode)) {
+            result.fail(new ClientErrorException(HttpURLConnection.HTTP_PRECON_FAILED,
+                    "server supports binding modes U and UQ only"));
+        } else {
+            final var registrationUpdate = new RegistrationUpdate(
+                    registrationId,
+                    EndpointContextUtil.extractIdentity(ctx.getExchange().advanced().getRequest().getSourceContext()),
+                    lifetime,
+                    null,
+                    bindingMode,
+                    objectlinks,
+                    null);
+            final var updatedRegistration = lwm2mRegistrationStore.updateRegistration(registrationUpdate);
+
+            sendLifetimeAsTtdEvent(ctx, updatedRegistration.getUpdatedRegistration(), currentSpan)
+                    .onSuccess(ok -> result.complete(ResponseCode.CHANGED))
+                    .onFailure(t -> result.fail(t));
+        }
+        return result.future()
+                .onFailure(t -> TracingHelper.logError(currentSpan, t))
+                .onComplete(r -> currentSpan.finish());
+    }
+
+    private Future<?> sendLifetimeAsTtdEvent(
+            final CoapContext ctx,
+            final Registration registration,
+            final Span currentSpan) {
+
+        final int lifetime = registration.getBindingMode() == BindingMode.U
+                ? registration.getLifeTimeInSec().intValue()
+                : 20;
+        return getAdapter().sendTtdEvent(
```

Comment:
OK, now I get the "lifetime" stuff. I'm not sure, if all devices will do that proper. In fact, sometimes the device itself stays "reachable", but k8s or other network-components closes the ip-route. We will see ...